src/ResourceManager/Resources/Commands.Resources/Models.ResourceGroups/ResourceIdentifier.cs

```diff
@@ -138,10 +138,18 @@ public override string ToString()
 
             AppendIfNotNull(ref resourceId, "/subscriptions/{0}", Subscription);
             AppendIfNotNull(ref resourceId, "/resourceGroups/{0}", ResourceGroupName);
-            AppendIfNotNull(ref resourceId, "/providers/{0}", provider);
-            AppendIfNotNull(ref resourceId, "/{0}", parentAndType);
-            AppendIfNotNull(ref resourceId, "/{0}", ResourceName);
 
+            // Only append parentAndType and ResourceName when provider is not empty.
+            if (!string.IsNullOrEmpty(provider))
+            {
+                resourceId.AppendFormat("/providers/{0}", provider);
+                if (!string.IsNullOrEmpty(parentAndType))
+                {
+                    resourceId.AppendFormat("/{0}", parentAndType);
+                    AppendIfNotNull(ref resourceId, "/{0}", ResourceName);
+                }
+            }
```

Comment:
I agree, but doing the validation later might not be able to give user the correct error message. e.g. for this issue, we tell user that the composed scope doesn't have even number of parts, but actually the root cause is user didn't have a type in the ResourceType. 
The ultimate goal should be validation on each input parameter (like this change we do on Scope), like i mentioned in other comment, it should be done in another work item.

---
In reply to: [118643432](https://github.com/Azure/azure-powershell/pull/4019#discussion_r118643432) [](ancestors = 118643432)